### PR TITLE
tests: add tests for a couple of specific URL openers

### LINKED ISSUE #1515: all: improve testing for URLOpeners
Today drivers test URLs by opening them and checking for error, but they don't really verify that the resource was opened with the correct arguments. Fix that.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #1515.

### CODE REVIEW

File: runtimevar/runtimeconfigurator/runtimeconfigurator.go [hunk 1]
```diff
@@ -213,6 +209,25 @@ type ResourceName struct {
 	Variable  string
 }
 
+func newResourceNameFromURL(u *url.URL) (ResourceName, error) {
+	var rn ResourceName
+	rn.ProjectID = u.Host
+	// Using SplitN because the variable name can be hierarchical; we
+	// take the first path element as the Config and the result for the
+	// Variable.
+	if pathParts := strings.SplitN(strings.TrimPrefix(u.Path, "/"), "/", 2); len(pathParts) == 2 {
```
Comment (jba) [inline runtimevar/runtimeconfigurator/runtimeconfigurator.go]: Line's a bit long; please split.

Comment (jba) [reply runtimevar/runtimeconfigurator/runtimeconfigurator.go]: pathParts := expr
if len(pathParts)...

File: secrets/azurekeyvault/akv_test.go [hunk 2]
```diff
@@ -157,6 +158,41 @@ func (v verifyAs) ErrorCheck(k *secrets.Keeper, err error) error {
...
+	for _, test := range tests {
+		u, err := url.Parse(test.URL)
+		if err != nil {
+			t.Fatal(err)
+		}
+		gotVault, gotKey, gotVersion, gotErr := keyInfoFromURL(u)
+		if (gotErr != nil) != test.WantErr {
+			t.Errorf("%s: got error %v, want error %v", test.URL, gotErr, test.WantErr)
+		}
+		if gotVault != test.WantVault {
```
Comment (jba) [inline secrets/azurekeyvault/akv_test.go]: Only do these three checks if gotErr == nil?

File: runtimevar/runtimeconfigurator/runtimeconfigurator.go (hunk 1 above)
Comment (vangent) [reply runtimevar/runtimeconfigurator/runtimeconfigurator.go]: Done.

File: secrets/azurekeyvault/akv_test.go (hunk 2 above)
Comment (vangent) [reply secrets/azurekeyvault/akv_test.go]: Doesn't hurt to do them, but OK; done.

**Pull request merged**